tests/lib/rules/no-restricted-imports.js

```diff
@@ -35,6 +35,15 @@ ruleTester.run("no-restricted-imports", rule, {
         {
             code: "import withGitignores from \"foo/bar\";",
             options: [{ patterns: ["foo/*", "!foo/bar"] }]
+        },
+        {
```

Comment:
Need a lot more unit tests here, to verify different import patterns, like importing multiple properties, renaming them, importing `*`, etc.

Reply:
Currently, I have only implemented this feature using paths, since patters does not currently support additional arguments.

If we want this implemented for both, I think it might make sense for me to do some refactoring first to make the patterns/paths options consistent

What are your thoughts, @ilyavolodin?

Currently:
```
items: {
  type: "object",
  properties: {
    paths: arrayOfStringsOrObjects,
    patterns: arrayOfStrings
  },
  additionalProperties: false
},
```

Reply:
Sorry, I'm not sure I understand this comment. All of the imports are based on paths. The patterns that I'm talking about are like this:
```js
import withGitignores as gitIgnore from "foo/bar";
import {withGitignores} from "foo/bar";
import * as gitIgnore from "foo/bar";
```
etc.

P.S. Oh, I see, you are talking about patterns as in configuration option for the rule. I was just talking about generic import statement patterns. But it's a good point, if we are adding this option, we should probably also introduce it for patterns (config option) as well. And I agree, that patterns should also accept one of either strings or objects.

Reply:
Sounds good.  I'll also work on converting the patterns (config option) to accept strings or objects.

Reply:
In my latest commit, I've added a lot more unit tests.  I used the syntax section of [mdn import statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) as a guide to create my scenarios.

Let me know if you can think of any more unit tests that would be valuable.

Reply:
Unit tests look a lot better now, thanks!